sql/driver/driver_test.go

```diff
@@ -363,3 +363,29 @@ CREATE TABLE t.kv (
 		t.Fatalf("expected %s, but got %s", expectedResults, results)
 	}
 }
+
+func TestInsecure(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	// Start test server in insecure mode.
+	s := &server.TestServer{}
+	s.Ctx = server.NewTestContext()
+	s.Ctx.Insecure = true
+	if err := s.Start(); err != nil {
+		t.Fatalf("Could not start server: %v", err)
+	}
+	t.Logf("Test server listening on %s: %s", s.Ctx.RequestScheme(), s.ServingAddr())
+	defer s.Stop()
+
+	// We can't attempt a connection through HTTPS since the client just retries forever.
+	// DB connection using plain HTTP.
+	db, err := sql.Open("cockroach", "http://root@"+s.ServingAddr())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		_ = db.Close()
+	}()
+	if _, err := db.Exec("CREATE DATABASE t"); err != nil {
```

Comment:
LGTM
On Jul 21, 2015 3:37 PM, "MBerhault" <notifications@github.com> wrote:

> In sql/driver/driver_test.go
> <https://github.com/cockroachdb/cockroach/pull/1755#discussion_r35144581>:
>
> > +	if err := s.Start(); err != nil {
> > +		t.Fatalf("Could not start server: %v", err)
> > +	}
> > +	t.Logf("Test server listening on %s: %s", s.Ctx.RequestScheme(), s.ServingAddr())
> > +	defer s.Stop()
> > +
> > +	// We can't attempt a connection through HTTPS since the client just retries forever.
> > +	// DB connection using plain HTTP.
> > +	db, err := sql.Open("cockroach", "http://root@"+s.ServingAddr())
> > +	if err != nil {
> > +		t.Fatal(err)
> > +	}
> > +	defer func() {
> > +		_ = db.Close()
> > +	}()
> > +	if _, err := db.Exec("CREATE DATABASE t"); err != nil {
>
> bah. create is enough, I just needed something to actually use the
> connection. I was going to use sql.Ping(), but that doesn't seem to do
> anything (looking)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1755/files#r35144581>.
>